torch/distributed/fsdp/fully_sharded_data_parallel.py

```diff
@@ -1576,6 +1522,23 @@ def _reset_lazy_init(self) -> None:
         # set 'self.reshard_after_forward' flag based on self.sharding_strategy
         self._init_reshard_after_forward()
 
+    def _init_reshard_after_forward(self):
+        if self.sharding_strategy == ShardingStrategy.FULL_SHARD:
+            # Free full params and keep shard only after forward
+            self.reshard_after_forward = True
+        elif self.sharding_strategy == ShardingStrategy.SHARD_GRAD_OP:
```

Comment:
simplify to `elif sharding_strategy in [shard_grad_op, no_shard]`?

Reply:
Sorry, I think I just moved this code. I remove `reshard_after_forward` in the prefetching PR, so I will leave this be.